h/db/__init__.py

```diff
@@ -44,31 +45,24 @@
 Session = sessionmaker()
 
 
-def init(engine, base=Base, should_create=False, should_drop=False, authority=None):
-    """Initialise the database tables managed by `h.db`."""
-    # Import models package to populate the metadata
-    import h.models  # pylint: disable=unused-import
-
-    if should_drop:  # pragma: no cover
-        # SQLAlchemy doesn't know about the report schema, and will end up
-        # trying to drop tables without cascade that have dependent tables
-        # in the report schema and failing. Clear it out first.
-        engine.execute("DROP SCHEMA IF EXISTS report CASCADE")
-        base.metadata.reflect(engine)
-        base.metadata.drop_all(engine)
-    if should_create:  # pragma: no cover
-        # In order to be able to generate UUIDs, we load the uuid-ossp
-        # extension.
-        engine.execute('CREATE EXTENSION IF NOT EXISTS "uuid-ossp";')
-        base.metadata.create_all(engine)
+def pre_delete(engine):  # pragma: no cover
+    engine.execute("DROP SCHEMA IF EXISTS report CASCADE")
+
+
+def pre_create(engine):  # pragma: no cover
+    engine.execute('CREATE EXTENSION IF NOT EXISTS "uuid-ossp";')
+
+
+def post_create(engine):  # pragma: no cover
+    authority = environ["AUTHORITY"]
```

Comment:
These are hooks that `init_db.py` calls to do h-specific stuff that needs to happen before/after creating/deleting the DB tables. It's a pain to have to add these but unfortunately there is h-specific stuff that needs to happen.